### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by rthijssen**
Hey all, I've had a go at updating the documentation based on what I've learned setting up Keymaster. Hope it's useful. Happy to fix anything I got wrong.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,51 +29,57 @@ You can use the prebuilt binaries at the [releases page](https://github.com/Syma
...
+* RPM (CentOS): Modify the following configuration items in your `config.yml` file:
+    * `data_directory: /var/lib/keymaster `
+    * `shared_data_directory: /usr/share/keymasterd/`.
+* DEB (Debian/Ubuntu): Modify the following configuration items in your `config.yml` file:
+    * `data_directory: /var/lib/keymaster `
+    * `shared_data_directory: /usr/share/keymasterd/`.
+
+##### Supported backend authentication methods
+Several authentication methods are supported by the `keymasterd` service. You can separately specify which authentication methods you accept for the web backend (`allowed_auth_backends_for_webui`) and for obtaining certificates (`allowed_auth_backends_for_certs`).
+* **LDAP**: For LDAP the `bind_pattern` is a printf string where `%s` is the place where the username will be substituted. For example for an 389ds/openldap string might be: `"uid=%s,ou=People,dc=example,dc=com`. To leverage LDAP authentication set the appropriate `allowed_auth_*` setting to `["federated"]`.
```
Comment (cviecco) [inline README.md]: The setting for ldap is password. ["federated"] is for using keymaster web-ui to be behind an oauth2 provider such as github or google.

File: README.md [hunk 2]
```diff
@@ -27,51 +29,57 @@ You can use the prebuilt binaries at the [releases page](https://github.com/Syma
...
+* **Admin Management Interface (default port 6920)**: The service exposed on port 6920 allows administrators or log collection systems to collect logs generated by the `keymasterd` service.
+
+To run `keymasterd` you will need to generate a config file. `keymasterd` facilitates this through the command-line arguments `-generateConfig` and `-alsoLogToStderr`. Running the `keymasterd` binary with these arguments will generate the following:
+* A configuration file. By default `keymasterd` will write this file to `/etc/keymaster/config.yml`.
+* The Keymaster CA key pair. The encrypted private key  (`masterkey.asc`) is an armored PGP file. For development (or if your trust model permits it) you can decrypt the private-key and write it to the filesystem. To decrypt the key run `gpg -ad $Filename`. Once decrypted set the `ssh_ca_filename` field in the `keymasterd` config file to the path of the decrypted master key.
+* Server keys (for Testing Purposes only): the `server.pem` and `server.key` files are generated to ????
```
Comment (cviecco) [inline README.md]: Testing self signed keys for localhost. This are needed as keymaster requires TLS

**Comment by rthijssen** [comment]
Ah, thanks for that. I actually noticed one small issue on line 51. Just not entirely sure what a new commit to this pull request would do to your updates.... Easier is probably to replace ` are generated to ????` with a full stop.

**Comment by cviecco** [comment]
I manually added the changes (via github editor). You usually do your changes and do another pull request. Thank you for your first commit

**Pull request merged**